coderd/database/queries/workspaces.sql

```diff
@@ -118,7 +119,7 @@ LEFT JOIN LATERAL (
 		provisioner_jobs.job_status
 	FROM
 		workspace_builds
-	LEFT JOIN
+	JOIN
```

Comment:
I would avoid making this change. I believe there's a window of time when there exists a new build but it's not assigned to a job. So the nullability is something that should be handled.

This logic change essentially results in returning the previous build until the job has started.

Reply:
I'm not sure if that's possible?

```sql
create table if not exists workspace_builds
(
    id                  uuid                                                                                not null
    ...
    workspace_id        uuid                                                                                not null
        references public.workspaces
            on delete cascade,
    ...
    job_id              uuid                                                                                not null
        unique
        references public.provisioner_jobs
            on delete cascade,
    ...
);
```

These keys are all non-nullable.
Is there something I'm missing?

Reply:
I should say though that I'd rather not make this change if you in any way suspect this will cause issues.
The way I had it prior to https://github.com/coder/coder/pull/12762#discussion_r1539514142 worked fine.

Reply:
Builds are inserted with their respective job in a single transaction.  c.f. `wsbuilder`

Also, Danny's right that job_id is `NOT NULL` with a foreign key constraint, so even if we broke `wsbuilder` it still wouldn't be possible to have a build with no job.

Reply:
Given the constraint, the change is perfectly fine 👍. I should’ve checked it myself before commenting. Sorry for the noise @dannykopping.

Reply:
The DB schema should probably be updated to include NOT NULL on the foreign keys, though. Otherwise NULL entries are not enforced by the DB.

Reply:
No worries @mafredri!

I just generated the DDL from an existing database in my IDE (GoLand). The above also has `not null` but you have to scroll over a bit to the right.

Reply:
Haha, it was perfectly hidden by the code block size 😂.